### PR TITLE
libmicrohttpd: Add new package

### PR DESCRIPTION
**Pull request by hainest**
This is needed to support building elfutils with debuginfod.

### CODE REVIEW

**Comment by hainest** [comment]
I don't understand which package is being referred to in the minimum version check failure. Could someone elaborate on that?

File: var/spack/repos/builtin/packages/libmicrohttpd/package.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            '--disable-rpath',      # let spack handle the RPATH
+            '--disable-doc',        # don't build the docs
+            '--disable-examples',   # don't build the examples
+            '--disable-curl'		# disable cURL-based testcases
+        ]
+
+        if self.spec.satisfies('+https'):
+            options.append('--enable-https')
+            prefix = self.spec['gnutls'].prefix
+            options.append('--with-gnutls={}'.format(prefix))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmicrohttpd/package.py]:
```suggestion
            options.append('--with-gnutls={0}'.format(prefix))
```
This syntax doesn't work in Python 2.6, hence the version compatibility check warning.

File: var/spack/repos/builtin/packages/libmicrohttpd/package.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            prefix = self.spec['libgcrypt'].prefix
+            options.append('--with-libgcrypt-prefix={}'.format(prefix))
+
+        return options
+
+    def install(self, spec, prefix):
+        make()
+        make('install')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmicrohttpd/package.py]: You can delete this, it's the default build and install phases

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libmicrohttpd/package.py (hunk 1 above)
Comment (hainest) [reply var/spack/repos/builtin/packages/libmicrohttpd/package.py]: Ah, ok. I couldn't deduce what the warning was.

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/libmicrohttpd/package.py (hunk 2 above)
Comment (hainest) [reply var/spack/repos/builtin/packages/libmicrohttpd/package.py]: It was part of the default template of `spack create`, so I wasn't sure if it was needed or not. Removed.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/libmicrohttpd/package.py]: Must have been a bug in an old version of Spack, it isn't added for me when it chooses `AutotoolsPackage`

**Review by alalazo — APPROVED** [review]

**Pull request merged**